core/node/node_framework/src/implementations/layers/web3_api/server.rs

```diff
@@ -0,0 +1,225 @@
+use std::num::NonZeroU32;
+
+use tokio::{sync::oneshot, task::JoinHandle};
+use zksync_core::api_server::web3::{state::InternalApiConfig, ApiBuilder, ApiServer, Namespace};
+
+use crate::{
+    implementations::resources::{
+        healthcheck::AppHealthCheckResource, pools::ReplicaPoolResource,
+        sync_state::SyncStateResource, web3_api::TxSenderResource,
+    },
+    service::{ServiceContext, StopReceiver},
+    task::Task,
+    wiring_layer::{WiringError, WiringLayer},
+};
+
+/// Set of optional variables that can be altered to modify the behavior of API builder.
+#[derive(Debug, Default)]
+pub struct Web3ServerOptionalConfig {
+    pub namespaces: Option<Vec<Namespace>>,
+    pub filters_limit: Option<usize>,
+    pub subscriptions_limit: Option<usize>,
+    pub batch_request_size_limit: Option<usize>,
+    pub response_body_size_limit: Option<usize>,
+    pub websocket_requests_per_minute_limit: Option<NonZeroU32>,
+    pub tree_api_url: Option<String>,
```

Comment:
Future work (?): It could make sense to make tree API client a resource. Could be done separately.

Reply:
Yup, also thought about it.